sound/soc/sof/intel/cnl.c

```diff
@@ -361,6 +364,10 @@ int sof_cnl_ops_init(struct snd_sof_dev *sdev)
 
 		/* ipc */
 		sof_cnl_ops.send_msg	= cnl_ipc4_send_msg;
+
+		/* dsp core get/put */
+		sof_cnl_ops.core_get = hda_dsp_ipc4_core_get;
+		sof_cnl_ops.core_put = hda_dsp_ipc4_core_put;
```

Comment:
let's enable multi-core for TGL+, we don't have a reason to spend time on multicore for earlier platforms at the moment, we already have a full plate with IPC4 in general.

Reply:
@plbossart, @ranj063, this PR takes a simplistic approach to tackle IMR boot by implementing and using the generic core_put() callback to let the firmware know that we are need to prepare for IMR.
It is a side effect that for multicore support we also need the core_put(), but if I'm not mistaken the actual multicore support depends on the loaded topology. Even if we have core_put() implemented for the platform it is not going to matter if the topology we are loading does not use multicore.
I did not wanted to implement another IPC4 wrapper for sending the SET_DX.
If we want to keep IMR, but leave out core_put() then I don't see other way around this. I don't think we want to disable IMR just because multicore is not stable on the given platform with SOF?

Reply:
@ranj063, what will fail with SOF.IPC4? the IPC message for non primary core or the host operation?

If we load a topology without multicore support and have the core_put() implemented, does that going to fail? DO you know if we have some platform flag to say that multicore is not supported or is it only that we don't support the core_put on some platforms?

We can try to remove support for IMR boot when core_put() is not implemented for IPC4, but would not it be a regression? APL with the closed source firmware works fine with IMR for example and if SOF.IPC4 would support APL, I would expect to work there as well. Afaik IPC3 on APL supports IMR boot as well, no?

Let me put this back to the drawing board a bit to figure out a clean way to separate the IMR and multicore (core_put) support.

Reply:
@ujfalusi if we don't enable multicore in tplg and do a core put for core 0 only.we.might be ok but something to try with ipc4 and SOF. Let me try it out today